virtualcluster/cmd/syncer/app/server.go

```diff
@@ -132,6 +132,25 @@ func Run(cc *syncerconfig.CompletedConfig, stopCh <-chan struct{}) error {
 		}
 	}()
 
+	go func() {
+		// start a pprof http server
+		klog.Fatal(http.ListenAndServe(":6060", nil))
+	}()
+
+	var leaderElector *leaderelection.LeaderElector
+	go func() {
+		// start a health http server.
+		mux := http.NewServeMux()
+		healthz.InstallHandler(mux)
+		healthz.InstallReadyzHandler(mux, healthz.NamedCheck("leader", func(r *http.Request) error {
+			if leaderElector != nil && !leaderElector.IsLeader() {
```

Comment:
This will probably cause issues during a rolling upgrade. Pods must transition to ‘Ready’ to allow the upgrade to proceed when managed by the Deployment controller.

Readiness shouldn't be used to indicate whether instances are leaders 😄

Reply:
Ok, I removed it for this PR. To be followed up later